### PR TITLE
Add exercise conclusions.

### PR DESCRIPTION
**Pull request by dvdstelt**
In two documents there was a link to Google Groups. Changed it to new Discourse platform.
One of those two documents contained a link at the bottom of the exercise, I added a conclusion with link to the bottom of all exercises to be consistent. It's a good thing to have the ability for attendees (or anyone just looking at this repo) to discuss the exercises and workshop content after the workshop.

### CODE REVIEW

**Comment by mauroservienti** [comment]
I learned from the master, so be prepared: the 2 commits should be into 2 different PRs 😄, one for the links update and the second one for the exercises conclusion.

The content is good to go, though

File: exercises/03-sagas/instructions.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ For more info, please see [the instructions for running the exercise solutions](
 
 At the moment we gather all successfully paid orders and the only option we have is to process them in batches. Business is growing rapidly and we would like to process orders as soon as possible, not just at predefined intervals.
 
+Note: This is a hypothetical case. From experience we've seen a lot of developers implement batch processes to verify a certain state in the database. Sagas are
```
Comment (adamralph) [inline exercises/03-sagas/instructions.md]: I haven't looked through the rest of the PR yet, but there's something missing here.

**Comment by adamralph** [comment]
I cherry-picked the fixing of the forum link into https://github.com/Particular/Workshop.Microservices/pull/154

This PR can now become the addition of exercise conclusions. Marking as WIP until #154 is merged.

**Comment by dvdstelt** [comment]
Didn't see the squash remark. Can we also squash it when merging into master? I still suck at Git, especially when it's been pushed.

File: exercises/01-composite-ui/README.md [hunk 2]
```diff
@@ -125,3 +125,9 @@ Update the `orders` list template (`app\branding\orders\ordersView.html`) to dis
 <br />
 <strong>Order total:</strong> {{order.orderTotalPrice}}
 ```
+
+## Conclusion
+
+In this exercise we've seen how to combine data from various autonomous services into a user interface, in which the end-user is unaware of the architecture behind it.
```
Comment (adamralph) [inline exercises/01-composite-ui/README.md]: > in which the end-user is unaware of the architecture behind it

This doesn't feel right.  I'd leave it out, and perhaps call out that there is a _single_ user interface:

> In this exercise we've seen how to combine data from various autonomous services into a single user interface.

File: exercises/02-publish-subscribe/README.md [hunk 3]
```diff
@@ -407,4 +407,10 @@ This is a powerful feature which could be of huge value to operations activities
...
+- You might notice several endpoints with the same name. Endpoints send heartbeats with a unique host identifier, made up of their endpoint name and a hash of the folder the endpoint is installed in. Our exercises all have the same endpoint name, but different folders. Another example is when you deploy endpoints using [Octopus](https://octopus.com/). This will deploy every version in its own folder, with the result that every version will spawn a new monitored endpoint in ServicePulse. You can solve this by [overriding the host identifier](https://docs.particular.net/nservicebus/hosting/override-hostid).
+
+## Conclusion
+
+Hopefully this exercise has demonstrated how you can use the publish/subscribe messaging pattern to decouple services.
```
Comment (adamralph) [inline exercises/02-publish-subscribe/README.md]: I would leave out "Hopefully":

> This exercise has demonstrated...

File: exercises/03-sagas/README.md [hunk 4]
```diff
@@ -32,6 +32,8 @@ For more info, please see [the instructions for running the exercise solutions](
 
 At the moment we gather all successfully paid orders and the only option we have is to process them in batches. Business is growing rapidly and we would like to process orders as soon as possible, not just at predefined intervals.
 
+Note: This is a hypothetical case. From experience we've seen a lot of developers implement batch processes to verify a certain state in the database. Sagas are a solution to batch processes. If you would like to know more, you can [read our blogpost on this](https://particular.net/blog/death-to-the-batch-job) at a later time.
```
Comment (adamralph) [inline exercises/03-sagas/README.md]: > ...to verify a certain state in the database

I'm not sure what is meant by this.

Comment (adamralph) [inline exercises/03-sagas/README.md]: > Sagas are a solution to batch processes.

Perhaps:

> Sagas are an alternative to batch processes.

Comment (adamralph) [inline exercises/03-sagas/README.md]: > If you would like to know more, you can [read our blogpost on this](https://particular.net/blog/death-to-the-batch-job) at a later time.

Suggest:

> For more info, read our blog post, ["Death to the batch job"](https://particular.net/blog/death-to-the-batch-job).

File: exercises/03-sagas/README.md [hunk 5]
```diff
@@ -264,3 +266,9 @@ Create a new saga in the Finance bounded context which first tries to process th
 Sagas are not supposed to retrieve data from a data store or call out to external systems. They should execute tasks using [the request/response pattern](http://docs.particular.net/nservicebus/sagas/#sagas-and-request-response). This means that, rather than processing the payment directly, the saga should send a message to another handler in the Finance bounded context to process the payment on its behalf. Whether it fails or succeeds, that handler should [reply to the saga](http://docs.particular.net/nservicebus/messaging/reply-to-a-message) with the status of the payment. If the payment failed, the saga should send another message to process the payment, but this time to the reliable, but expensive, provider.
 
 The solution should end up with at least two new handlers and a new saga. Depending on your solution, you may end up with more.
+
+## Conclusion
+
+Hopefully this exercise has demonstrated how to use sagas to orchestrate business processes.
```
Comment (adamralph) [inline exercises/03-sagas/README.md]: Again, I would omit "Hopefully".

File: exercises/04-integration/README.md [hunk 6]
```diff
@@ -426,6 +426,8 @@ Consider the following items and perhaps discuss them with your colleagues:
 - How did you deal with those? Did you have to run batch jobs, or connect directly to the database to hack the data manually? Or some other complicated workaround?
 - When this happened? Were you able to ask the business how to handle the failures and then design the system to match the solution described by the business? Or did you have to translate the business language into a complex set of technical operations which approximated the business language?
 
-Hopefully this exercise has demonstrated how powerful sagas can be to orchestrate business processes.
+## Conclusion
 
-If you'd like to discuss this more, please don't hesitate to drop us a line in our [community discussion forum](https://groups.google.com/forum/#!forum/particularsoftware).
+Hopefully this exercise has demonstrated how to act on integration issues, where you have the same issue as in the user-interface where data from multiple services need to be brought together. We also saw again how powerful sagas can be to orchestrate business processes.
```
Comment (adamralph) [inline exercises/04-integration/README.md]: Suggest:

> This exercise has demonstrated how to address integration requirements where, as with the UI, data from multiple services needs to be combined in order to perform some action. We also saw, again, how powerful sagas can be in orchestrating business processes.

File: exercises/02-publish-subscribe/README.md (hunk 3 above)
Comment (adamralph) [inline exercises/02-publish-subscribe/README.md]: Nitpick: "publish/subscribe" should be "Publish-Subscribe" to be consistent with the rest of the document.

(There is one other existing inconsistency, and also we should lower case the term throughout, but that is something for another PR.)

**Comment by dvdstelt** [comment]
Please review again

File: exercises/03-sagas/README.md [hunk 7]
```diff
@@ -32,6 +32,8 @@ For more info, please see [the instructions for running the exercise solutions](
 
 At the moment we gather all successfully paid orders and the only option we have is to process them in batches. Business is growing rapidly and we would like to process orders as soon as possible, not just at predefined intervals.
 
+Note: This is a hypothetical case. From experience we've seen a lot of developers implement a batch processes. Sagas are a better alternative to batch processes. For more info, read our blog post, "[Death to the batchjob](https://particular.net/blog/death-to-the-batch-job)".
```
Comment (adamralph) [inline exercises/03-sagas/README.md]: - "...implement a batch processes" - remove "a"
- "batchjob" - should be "batch job"

**Comment by mauroservienti** [comment]
yes

**Pull request merged**